ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx

```diff
@@ -288,7 +294,7 @@ function WorkloadCvesOverviewPage() {
         // Reset all filters, sorting, and pagination and apply to the current history entry
         setActiveEntityTabKey('CVE');
         setSearchFilter({});
-        sort.setSortOption(getDefaultWorkloadSortOption('CVE'));
+        sort.setSortOption(getDefaultWorkloadSortOption('CVE', searchFilter));
```

Comment:
Help my mental fail whether `searchFilter` here is `{}` following `setSearchFilter({});` call.

Reply:
Not a mental fail, but a mental win. This actually causes a bug when a severity filter is applied on the Observed tab, and the user moves to either Deferred or False positive. The wrong default sort will be applied.

In fact, my understanding of this wasn't correct originally. Any time we request the default sort we would be doing so with an empty `searchFilter`, so the additional processing in `getDefaultWorkloadSortOption` is unnecessary. I'll revert this change.

Reply:
Ok got it - we _do_ need the ability to pass the searchFilter, but not in the previous location. This needs to occur when the user directly visits the page with applied severity filters that do not match the default filters.

There was a test that checked this scenario for default filters, however when the defaults are applied that triggers a re-calculation of the necessary sort. This does not occur when the user visits directly via a link that included severity filters. I added an additional check in the existing test to cover this case.